lib/stub_ui/lib/src/ui/text.dart

```diff
@@ -1011,6 +1011,185 @@ enum BoxWidthStyle {
+  ) : runMetrics = Map() {
+    for (int i = 0; i < rMetrics.length; ++i) {
+      runMetrics[rMetricsIndexes[i]] = rMetrics[i];
+    }
+  }
+
+  /// The index in the text buffer the line begins.
+  final int startIndex;
+
+  /// The index in the text buffer the line ends.
+  final int endIndex;
+
+  /// The index in the text buffer the line ends ignoring whitespace.
+  ///
+  /// This value is provided as if any whitespace at the end were truncated.
+  final int endExcludingWhitespace;
+
+  /// The index in the text buffer the line ends including newline characters.
+  ///
+  /// This value is provided as if any trailing newline characters for the
+  /// current line were included. Newline characters in other lines are always
+  /// accounted for.
+  final int endIncludingNewline;
+
+  /// True if this line ends with a explicit line break (eg, '\n') or is the end
+  /// of the paragraph. False otherwise.
+  final bool hardBreak;
+
+  // The final computed ascent and descent for the line. This can be impacted by
+  // the strut, height, scaling, as well as outlying runs that are very tall.
+  //
+  // The top edge is `baseline - ascent` and the bottom edge is `baseline +
+  // descent`. Ascent and descent are provided as positive numbers. Raw numbers
+  // for specific runs of text can be obtained in run_metrics_map. These values
+  // are the cumulative metrics for the entire line.
+
+  /// The rise from the [baseline] as calculated from the font and style for this line.
+  ///
+  /// The [ascent] is provided as a positive value, even though it is typically defined in
+  /// fonts as negative. This is to ensure the signage of operations with these
+  /// metrics directly reflects the intended signage of the value. For example,
+  /// the y-coordinate of the top edge of the line is 'baseline - ascent`.
+  final double ascent;
+
+  /// The drop from the [baseline] as calculated from the font and style for this line.
+  ///
+  /// The y-coordinate of the bottom edge of the line is 'baseline + descent`.
+  final double descent;
+
+  /// The rise from the [baseline] as calculated from the font and style for this line
+  /// ignoring the text scale factor.
+  ///
+  /// The [unscaledAscent] is provided as a positive value, even though it is typically
+  /// defined in fonts as negative. This is to ensure the signage of operations with
+  /// these metrics directly reflects the intended signage of the value. For example,
+  /// the y-coordinate of the top edge of the line is 'baseline - ascent`.
+  final double unscaledAscent;
+
+  /// Total height of the line from the top edge to the bottom edge.
+  ///
+  /// This is equivalent to `ascent + descent`
+  final double height;
+
+  /// Width of the line.
+  final double width;
+
+  /// The x-coordinate of left edge of the line.
+  ///
+  /// The right edge can be obtained with `left + width`.
+  final double left;
+
+  /// The y position of the baseline for this line from the top of the paragraph.
```

Comment:
We should be consistent in our style here: `x-coordinate`, `y position`, etc. Maybe see what we do today in the framework (e.g. it might be `_x_ coordinate")